### PR TITLE
added recipe for asap

### PR DESCRIPTION
**Pull request by ThomasWollmann**
* [x] I have read the guidelines above.
* [x] This PR adds a new recipe.
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

**Comment by ThomasWollmann** [comment]
depends on https://github.com/bioconda/bioconda-recipes/pull/3432 and https://github.com/bioconda/bioconda-recipes/pull/3436

File: recipes/asap/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+{% set ASAP_VERSION = "1.6" %}
+package:
+    name: asap
+    version: {{ ASAP_VERSION }}
+
+source:
+    fn: {{ ASAP_VERSION }}.zip
+    url: https://github.com/GeertLitjens/ASAP/archive/{{ ASAP_VERSION }}.zip
```
Comment (druvus) [inline recipes/asap/meta.yaml]: Missing checksum

Comment (ThomasWollmann) [inline recipes/asap/meta.yaml]: Checksum changes each version. Should I add it for the current version?

Comment (bgruening) [inline recipes/asap/meta.yaml]: Yes, for the exact version your are downloading we need a checksum to verify that the download was successful and no harm was done with the tarball.

File: recipes/asap/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        - jpeg
+        - jasper
+        - dcmtk
+        - swig
+        - openslide
+        - pugixml
+        - zlib
+
+    run:
+        - python
```
Comment (bgruening) [inline recipes/asap/meta.yaml]: libgcc and these guidelines: https://bioconda.github.io/guidelines.html#c-c
there is also a section about boost and here an example: https://github.com/bioconda/bioconda-recipes/blob/master/recipes/salmon/meta.yaml#L25

**Comment by bgruening** [comment]
I feel your pain @ThomasWollmann! Not sure the latest error message is wired for CMAKE. Do you need to run it from the root dir?

**Comment by ThomasWollmann** [comment]
Maybe...

**Comment by bgruening** [comment]
Seems like this is the error: https://travis-ci.org/bioconda/bioconda-recipes/jobs/206493010#L646

**Comment by ThomasWollmann** [comment]
@bgruening This package is extremely tenacious...

**Comment by bgruening** [comment]
The latest error indicates a problem with dcmtk, I think? Do we need shared libs? https://github.com/bioconda/bioconda-recipes/pull/4207

Or the otherway around, do we need to create a static build for asap?

**Comment by ThomasWollmann** [comment]
The build instructions on https://github.com/GeertLitjens/ASAP

**Comment by ThomasWollmann** [comment]
Ops closed it by mistake.

**Comment by ThomasWollmann** [comment]
Seems like libijg8 in DCMTK is still build in a wrong way.

**Comment by bgruening** [comment]
There are no *.so files in DCMTK :(

**Comment by ThomasWollmann** [comment]
Just see *.a files...

**Comment by ThomasWollmann** [comment]
https://github.com/bioconda/bioconda-recipes/pull/4217

**Comment by ThomasWollmann** [comment]
@bgruening Now the *.so files should be available.

**Comment by bgruening** [comment]
`_POSITION_INDEPENDENT_CODE` ... OO :) Let's see. Otherwise, we need to convince asap to compile statically.

**Comment by bgruening** [comment]
```
/opt/conda/conda-bld/asap_vis_1509357862172/work/ASAP-1.7.3/ASAP/../imgproc/basicfilters/FilterBase.h:4:33: fatal error: basicfilters_export.h: No such file or directory
 #include "basicfilters_export.h"
```

I'm wondering if this tool is working at all :(

**Comment by ThomasWollmann** [comment]
The tool is popular in histology image analysis research. I'm not sure if people just use the provided binaries or compile it themselves...

**Comment by ThomasWollmann** [comment]
Seems like everything from /imgproc is not included in the other compile tasks.

**Comment by johanneskoester** [comment]
We have migrated our build system from Travis CI to Circle CI. If you would like to continue working on this PR, please update with latest upstream master and reopen this PR.

**Pull request closed without merging**